ext/ForwardDiffGPUArraysCoreExt.jl

```diff
@@ -0,0 +1,62 @@
+module ForwardDiffGPUArraysCoreExt
+
+using GPUArraysCore: AbstractGPUArray
+using ForwardDiff: ForwardDiff, Dual, Partials, npartials, partials
+
+function ForwardDiff.seed!(duals::AbstractGPUArray{Dual{T,V,N}}, x,
+                           seed::Partials{N,V} = zero(Partials{N,V})) where {T,V,N}
+    idxs = collect(ForwardDiff.structural_eachindex(duals, x))
+    duals[idxs] .= Dual{T,V,N}.(x[idxs], Ref(seed))
+    return duals
+end
+
+function ForwardDiff.seed!(duals::AbstractGPUArray{Dual{T,V,N}}, x,
+                           seeds::NTuple{N,Partials{N,V}}) where {T,V,N}
+    idxs = collect(ForwardDiff.structural_eachindex(duals, x))[1:N]
+    duals[idxs] .= Dual{T,V,N}.(x[idxs], seeds[1:N])
+    return duals
+end
+
+function ForwardDiff.seed!(duals::AbstractGPUArray{Dual{T,V,N}}, x, index,
+                           seed::Partials{N,V} = zero(Partials{N,V})) where {T,V,N}
+    offset = index - 1
+    idxs = collect(Iterators.drop(ForwardDiff.structural_eachindex(duals, x), offset))
+    duals[idxs] .= Dual{T,V,N}.(x[idxs], Ref(seed))
+    return duals
+end
+
+function ForwardDiff.seed!(duals::AbstractGPUArray{Dual{T,V,N}}, x, index,
+                           seeds::NTuple{N,Partials{N,V}}, chunksize = N) where {T,V,N}
+    offset = index - 1
+    idxs = collect(
+        Iterators.drop(ForwardDiff.structural_eachindex(duals, x), offset)
+    )[1:chunksize]
+    duals[idxs] .= Dual{T,V,N}.(x[idxs], seeds[1:chunksize])
+    return duals
+end
+
+# gradient
+function ForwardDiff.extract_gradient!(::Type{T}, result::AbstractGPUArray,
+                                       dual::Dual) where {T}
+    # this closure is needed for gpu compilation
+    partial_fn(dual, i) = partials(T, dual, i)
```

Comment:
Since it appears multiple times and since you're closing over `dual` as well, I'd suggest defining it as a callable struct, something like
```julia
struct PartialsFn{T,D}
    dual::D
end
PartialsFn{T}(dual) where {T} = PartialsFn{T,typeof(d)}(d) 

(f::PartialsFn{T})(i) where {T} = partials(T, f.dual, i)
```
outside of this function.